lib/node_modules/@stdlib/stats/nanmean/lib/main.js

```diff
@@ -0,0 +1,101 @@
+/**
+* @license Apache-2.0
+*
+* Copyright (c) 2025 The Stdlib Authors.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*    http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+'use strict';
+
+// MODULES //
+
+var dtypes = require( '@stdlib/ndarray/dtypes' );
+var gnanmean = require( '@stdlib/stats/base/ndarray/nanmean' );
+var dnanmean = require( '@stdlib/stats/base/ndarray/dnanmean' );
+var snanmean = require( '@stdlib/stats/base/ndarray/snanmean' );
+var factory = require( '@stdlib/ndarray/base/unary-reduce-strided1d-dispatch-factory' );
+
+
+// VARIABLES //
+
+var idtypes = dtypes( 'real_and_generic' );
+var odtypes = dtypes( 'real_and_generic' );
+var policies = {
+	'output': 'same',
```

Comment:
```suggestion
var odtypes = dtypes( 'real_floating_point_and_generic' );
var policies = {
	'output': 'real_floating_point_and_generic',
```

"same" obviously cannot be true, as you can provide an integer array, but computing the mean is not guaranteed to return an integer value. Hence, if I provide an `int8`, I should **not** be returning an `int8`, as the result is very likely to be truncated.